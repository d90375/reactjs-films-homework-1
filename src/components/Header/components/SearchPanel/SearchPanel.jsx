import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './SearchPanel.scss';

class SearchPanel extends Component {
  state = { value: '' };

  handleChange = async (event) => {
    const query = event.target.value;
    this.setState({ value: query });
    const { setMoviesCondition, fetchMovies, genres } = this.props;

    if (query !== '') {
      await setMoviesCondition('search');
      const { condition } = this.props;
      fetchMovies(condition, genres, query);
    } else {
      await setMoviesCondition('tranding');
      const { condition } = this.props;
      fetchMovies(condition, genres);
    }
  }

  render() {
    const { value } = this.state;
    return (
      <input
        className={styles.input}
        type="search"
        value={value}
        onChange={this.handleChange}
      />
    );
  }
}

SearchPanel.propTypes = {
  setMoviesCondition: PropTypes.func.isRequired,
  fetchMovies: PropTypes.func.isRequired,
  genres: PropTypes.string.isRequired,
  condition: PropTypes.string.isRequired,
};

export default SearchPanel;
