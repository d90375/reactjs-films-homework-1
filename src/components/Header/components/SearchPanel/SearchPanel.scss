@mixin outline-none {
  &:focus,
  &:active {
    outline: none;
  }
}

.search {
  display: flex;
  align-items: flex-end;

  .input {
    width: 250px;
    height: 35px;
    padding: 0 20px;
    font-size: 14px;
    letter-spacing: 1px;
    line-height: 20px;
    font-family: "Helvetica Neue", Helvetica, sans-serif;
    color: #fff;
    background-color: transparent;
    border: 2px solid #fff;
    border-right: none;
    border-radius: 10px 0 0 10px;

    @include outline-none();

    @media (max-width: 480px) {
      width: 170px;
    }
  }

  .submit {
    display: inline-block;
    width: 35px;
    height: 35px;
    color: transparent;
    background-color: transparent;
    background-image: url("../../../../assets/images/search-icon.png");
    background-repeat: no-repeat;
    background-position: 50% 50%;
    background-size: 20px 20px;
    border: 2px solid #fff;
    border-left: none;
    border-radius: 0 10px 10px 0;
    cursor: pointer;

    @include outline-none();

    &:hover {
      background-color: rgba(0, 0, 0, 0.1);
    }
  }
}
