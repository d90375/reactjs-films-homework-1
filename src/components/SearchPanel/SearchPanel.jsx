import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './SearchPanel.scss';

class SearchPanel extends Component {
  state = {
    value: '',
  };

  handleChange = async (e) => {
    this.setState({ value: e.target.value });
  }

  render() {
    const { value } = this.state;
    const { getSearch } = this.props;

    return (
      <input
        className={styles.input}
        type="search"
        value={value}
        onChange={(e) => {
          this.handleChange(e);
          getSearch(e);
        }}
      />
    );
  }
}

SearchPanel.propTypes = {
  getSearch: PropTypes.func.isRequired,
};

export default SearchPanel;
