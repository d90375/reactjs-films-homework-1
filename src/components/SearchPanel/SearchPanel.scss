@mixin outline-none {
  &:focus,
  &:active {
    outline: none;
  }
}

.search {
  display: flex;
  align-items: flex-end;

  .search__input {
    width: 250px;
    height: 35px;
    padding: 0 20px;
    font-size: 14px;
    letter-spacing: 1px;
    line-height: 20px;
    font-family: "Helvetica Neue", Helvetica, sans-serif;
    color: #fff;
    background-color: transparent;
    border: 2px solid #fff;
    border-right: none;
    border-radius: 10px 0 0 10px;

    &[type="search"]::-webkit-search-decoration,
    &[type="search"]::-webkit-search-cancel-button,
    &[type="search"]::-webkit-search-results-button,
    &[type="search"]::-webkit-search-results-decoration {
      -webkit-appearance: none;
    }

    @include outline-none();

    @media (max-width: 480px) {
      width: 170px;
    }
  }

  .search__submit {
    display: inline-block;
    width: 35px;
    height: 35px;
    color: transparent;
    background-color: transparent;
    background-image: url("assets/images/search-icon.png");
    background-repeat: no-repeat;
    background-position: 50% 50%;
    background-size: 20px 20px;
    border: 2px solid #fff;
    border-left: none;
    border-radius: 0 10px 10px 0;
    cursor: pointer;

    @include outline-none();

    &:hover {
      background-color: rgba(0, 0, 0, 0.1);
    }
  }
}
