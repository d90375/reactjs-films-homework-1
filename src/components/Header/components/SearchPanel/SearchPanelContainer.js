import { connect } from 'react-redux';
import { getSearchPending, getSearchError } from '../../../../modules/search/searchSelector';
import { getGenres } from '../../../../modules/genres/genresSelector';
import { fetchSearch, setSearchQuery } from '../../../../modules/search/searchAction';
import SearchPanel from './SearchPanel';

const mapStateToProps = (state) => ({
  error: getSearchPending(state),
  pending: getSearchError(state),
  genres: getGenres(state),
});

const mapDispatchToProps = {
  fetchSearch,
  setSearchQuery,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchPanel);
