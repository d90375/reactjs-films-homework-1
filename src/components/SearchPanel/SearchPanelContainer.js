import { connect } from 'react-redux';
import { setMoviesCondition, fetchMovies } from '../../modules/movies/moviesAction';
import { getMoviesCondition } from '../../modules/movies/moviesSelector';
import { getGenres } from '../../modules/genres/genresSelector';
import SearchPanel from './SearchPanel';

const mapStateToProps = (state) => ({
  condition: getMoviesCondition(state),
  genres: getGenres(state),
});

const mapDispatchToProps = {
  setMoviesCondition,
  fetchMovies,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchPanel);
