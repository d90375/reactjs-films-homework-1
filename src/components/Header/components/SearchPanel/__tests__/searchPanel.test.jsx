import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create, act } from 'react-test-renderer';

import SearchPanel from '../SearchPanel';

test('SearchPanel renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<SearchPanel />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});

test('onChange function change input value', () => {
  const tree = create(
    <SearchPanel />,
  );

  const input = tree.root.findByProps({ type: 'search' });
  expect(input.props.value).toBe('');

  const e = { target: { value: 'hi' } };
  const mockFunction = jest.fn(() => input.props.onChange(e));

  act(() => {
    mockFunction();
  });

  expect(input.props.value).toBe('hi');
});
