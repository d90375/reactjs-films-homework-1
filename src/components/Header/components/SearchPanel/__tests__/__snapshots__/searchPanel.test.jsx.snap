// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel renders correctly 1`] = `
<form
  className="search"
>
  <input
    aria-label="search"
    className="input"
    onChange={[Function]}
    type="search"
    value=""
  />
  <input
    className="submit"
    type="submit"
  />
</form>
`;
